R/data.table.R

```diff
@@ -1153,7 +1153,8 @@ replace_dot_alias = function(e) {
             }
           }
           names(jsub)=""
-          jsub[[1L]]=as.name("list")
+          # dont wrap the RHS in list if it is a singular NULL and if not creating a new column
+          if (length(jsub[-1L]) == 1L && as.character(jsub[-1L]) == 'NULL' && all(lhs %chin% names_x)) jsub[[1L]]=as.name("identity") else jsub[[1L]]=as.name("list")
```

Comment:
@joshhwuu reading a bit more carefully here, part of the issue is relying on literal `NULL` being used (as opposed to using `null_variable` where `null_variable=NULL`). At a minimum, we should check the actual value of `j[[1L]]`, see https://github.com/markfairbanks/tidytable/pull/831.